mdsuite/calculators/green_kubo_ionic_conductivity.py

```diff
@@ -155,11 +167,14 @@ def _apply_operation(self, ensemble, index):
         -------
         MSD of the tensor_values.
         """
-        jacf = sum([signal.correlate(ensemble[:, idx], ensemble[:, idx],
-                                     mode="full",
-                                     method='auto') for idx in range(3)])
-        self.jacf += jacf[int(self.data_range - 1):]
-        self.sigma.append(np.trapz(jacf[int(self.data_range - 1):], x=self.time))
+        jacf = self.data_range * tf.reduce_sum(
+            tfp.stats.auto_correlation(ensemble,
+                                       normalize=False,
+                                       axis=0,
+                                       center=False), axis=-1)
+        self.jacf += jacf
+        self.sigma.append(np.trapz(jacf[:self.integration_range],
```

Comment:
`tfp.math.trapz` could be used, so we can eventually move everything into graph mode with tf.function

Reply:
This is only possible in the newest version of tfp which is not yet full deployed for all architectures.